pkg/overlay/overlay_linux.go

```diff
@@ -61,6 +63,28 @@ func MountWithOptions(contentDir, source, dest string, opts *Options) (mount spe
 		}
 		if stat, ok := st.Sys().(*syscall.Stat_t); ok {
 			if err := os.Chown(upperDir, int(stat.Uid), int(stat.Gid)); err != nil {
```

Comment:
(Pre-existing: Should all of these modifications of `upperDir` happen if the caller specified a custom (and thus, maybe, pre-populated) `UpperDirOptionFragment`?)